test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -332,3 +488,152 @@ func checkData(t *testing.T, start time.Time, lineCount int) {
 	)
 	assert.NoError(t, err)
 }
+
+func ValidateEntity(t *testing.T, logGroup, logStream string, end *time.Time, expectedEntity expectedEntity) {
+	log.Printf("Validating entity for log group: %s, stream: %s", logGroup, logStream)
+
+	logGroupInfo, err := getLogGroup()
+	for _, lg := range logGroupInfo {
```

Comment:
Is there no way to get a specific log group when you provide the name? 

Seems expensive to get all log groups in an account. We have a ton in our test account..

Reply:
I don't think there's a way to just search for one log group (why not is beyond me..) but I updated this so it passes in the log group as a prefix to search by (commit [006a057](https://github.com/aws/amazon-cloudwatch-agent-test/pull/433/commits/006a057bf450c4f0f243278f50bde941a01dd2d9))